napari/layers/points/points.py

```diff
@@ -928,6 +937,63 @@ def on_mouse_release(self, event):
                 self.selected_data = []
             self._set_highlight(force=True)
 
+    def _transform_color(
+        self, colors: ColorType, elem_name: str, default: str
+    ) -> np.ndarray:
+        """Helper method to return a ColorArray from an arbitrary user input."""
+        try:
+            transformed = transform_color(colors)
+        except (AttributeError, ValueError, KeyError):
+            warnings.warn(
+                f"The provided {elem_name} parameter contained illegal values, "
+                f"reseting all {elem_name} values to {default}."
+            )
+            transformed = transform_color(default)
+        else:
+            if (len(transformed) != 1) and (
+                len(transformed) != len(self.data)
+            ):
+                warnings.warn(
+                    f"The provided {elem_name} parameter has {len(colors)} entries, "
+                    f"while the data contains {len(self.data)} entries. Setting {elem_name} to {default}."
+                )
+                transformed = transform_color(default)
+        return transformed
+
+    def _tile_colors(self, colors: ColorType) -> np.ndarray:
+        """Takes an input color array and forces into being the length of self.data.
+        Used when a single color is supplied for many input points, but we need
+        self.face_color \\ self.edge_color to have the shape of the actual data.
+
+        Parameters
+        --------
+        color : ColorType
+            The user's input after being normalized by self._transform_color
+
+        Returns
+        -----
+        tiled : np.ndarray
+            A tiled version (if needed) of the original input
+        """
+        data_len = len(self.data)
+        # len == 0 data is handled somewhere else
+        if (len(colors) == data_len) or (data_len == 0):
+            return np.asarray(colors)
+        # If the user has supplied a list of colors, but its length doesn't
+        # match the length of the data, we warn them and return a single
+        # color for all inputs
+        if len(colors) != 1:
+            warnings.warn(
+                f"The number of supplied colors mismatch the number of given"
+                f" data points. Length of data is {data_len}, while the number of colors"
+                f" is {len(colors)}. Color for all points is resetted to white."
```

Comment:
My thought was that the user was actually trying to input more than one color - i.e. they're probably thinking they're doing it OK. I don't raise an exception here since the user was "obviously" not trying to be malicious, they're just incompetent :)

Reply:
Sure, but then something like keeping the first color, or keeping the existing set color, makes more sense to me than resetting to white. But I think this is edge case behavior so I'm not too worried about it, I guess.